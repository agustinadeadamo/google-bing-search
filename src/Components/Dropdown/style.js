/**
 * @desc Dependencies
 */
import styled from "styled-components";

/**
 * @desc Variables
 */
import { Colors } from 'Mainstyles/Variables';

export const Select = styled.select`
    width: 100%;
    height: 36px;
    padding-left: 5px;
    padding-right: 30px;
    border: 1px solid ${Colors.grey};
    font-size: 14px;
    margin: 10px 0;

    option {
        width: 100%;
    }
`;