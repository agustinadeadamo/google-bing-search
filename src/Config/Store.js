/**
 * @desc Dependencies
 */
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

/**
 * @desc Middlewares
 */
import Middlewares from './Middlewares';

/**
 * @desc Reducers
 */
import Reducers from 'Reducer/Reducer';

/**
 * @desc Store
 */
const Store = createStore(Reducers, composeWithDevTools(applyMiddleware(...Middlewares)));

export default Store;
