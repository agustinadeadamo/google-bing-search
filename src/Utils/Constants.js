/* Search mode */
export const SearchMode = {
    "GOOGLE" : 'Google',
    "BING" : 'Bing',
    "GOOGLE_BING" : 'Goolge & Bing',
};

export const MessageError = {
    'SERVER_ERROR': 'There was an error in the server',
    'APP_ERROR': 'There was an error in the app',
};

export const Apikeys = {
    'BING_KEY' : '1ef3364ab9ce48b1b180b109e6aab6fb',
    'GOOGLE_KEY': 'AIzaSyB5Mqgm1gE_sAvmQToGkD2fGnOw-Pl1m5w',
    'GOOGLE_CX': '18ca2e163998cf02d'
}

export const Endpoints = {
    'GOOGLE_SEARCH' : 'https://www.googleapis.com/customsearch/v1',
    'BING_SEARCH': 'https://api.bing.microsoft.com/v7.0/search'
}

