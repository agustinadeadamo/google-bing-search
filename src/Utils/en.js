export const English = {
    'ADD_VALUE_TO_SEARCH' : 'Add a value to search',
    'SELECT_OPTION' : 'Select an option',
    'SUBMIT' : 'Submit',
    'GOOGLE_BING_SEARCH' : 'Google & Bing Search',
}