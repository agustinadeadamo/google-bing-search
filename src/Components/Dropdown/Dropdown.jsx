/**
 * @desc Dependencies
 */
import React from "react";
import PropTypes  from 'prop-types';
import { useDispatch } from "react-redux";

/**
 * @desc Style Components
 */
import { Select } from './style';

/**
 * @desc Actions
 */
import { changeShowError } from 'Actions/ErrorActions';

/**
 * @desc Utils
 */
import { MessageError } from 'Utils/Constants';

const DropDown = (props) => {

    // Destructuring props
    const {
        data,
        disabled,
        selected,
        label
    } = props;

    // Store
    const dispatch =  useDispatch()

    /**
     * @desc Event that get's value on onchange.
     * 
     * @param { EventTarget } event
     * 
     */
	const onChangeSelected = (event) => {

        try {
        
            if( event.target.value !== undefined && event.target.value !== ""){
                // Gets value selected
                const { value = "" } = event.target;              
                // Sends value selected
                props.onChange(value);    
            }

        } catch(error) {
            dispatch(changeShowError(MessageError.APP_ERROR));
        }

    }

    return (
        <div data-testid="dropdown-container">

            <label>
                {label}
            </label>

            <Select  
            data-testid="dropdown-component"
            value={selected}
            disabled={disabled}
            onChange={newSelected => onChangeSelected(newSelected)}
            onKeyDown={ e =>  props.onKeyDown && props.onKeyDown(e)}>
                {
                Array.isArray(data) && data.map((option) => {
                    return (<option data-testid={option.value} value={option.value} key={option.value}>{option.text}</option>);
                })
                }
            </Select>
        </div>
    
    )

}

DropDown.propTypes = {
    selected: PropTypes.string.isRequired,
    data: PropTypes.array.isRequired,
    onKeyDown: PropTypes.func,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    label: PropTypes.string
}

DropDown.defaultProps = {
  selected: '',
  data: [],
  onChange: () => {},
  disabled: false,
  label: '',
  onKeyDown: () => {}
};

export default DropDown;
