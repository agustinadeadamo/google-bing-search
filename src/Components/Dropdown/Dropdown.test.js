/**
 * @desc Dependencies
 */
import { create } from 'react-test-renderer';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux'

/**
 * @desc Config
 */
import Store from 'Config/Store';

/**
 * @desc Utils
 */
import { English } from 'Utils/en';

/**
 * @desc Components
 */
import Dropdown from './Dropdown';

describe('[Dropdown Component]',() => {
  
    it('Renders correctly', () => {

        const {queryByTestId} = render(
            <Provider store={Store}>
                <Dropdown onChangeValue={() => {}} value=""/>
            </Provider>
        );

        expect(queryByTestId('dropdown-container')).toBeTruthy();

    });

    it('Matches snapshot', () => {
        
        const dropdown = create(
            <Provider store={Store}>
                <Dropdown onChangeValue={() => {}} value=""/>
            </Provider>
        ).toJSON;

        expect(dropdown).toMatchSnapshot();

    });

    it('Updates value on change', () => {

        // Select options
        let options = [
            {
                text: '1',
                value: 'val1'
            },
            {
                text: '2',
                value: 'val2'
            }
        ]

        // Mocks function
        const onChangeSelected = jest.fn();

        const {queryByTestId, getByTestId} = render(
            <Provider store={Store}>
                <Dropdown data={options} onChange={onChangeSelected} value="" />
            </Provider>
        );
        
        // Gets dropdown component
        const dropdownComponent = queryByTestId('dropdown-component');

        // Generates change event
        fireEvent.change(dropdownComponent, {target: {value: 'val1'}});

        expect(getByTestId("val1").selected).toBe(true);
        expect(getByTestId("val2").selected).toBe(false);

    });

    it('Shows correct label', () => {  
        
        const { getByText} = render(
            <Provider store={Store}>
                <Dropdown  label={English.SELECT_OPTION} onChange={() => {}} value="" />
            </Provider>
        );

        expect(getByText(English.SELECT_OPTION)).toBeInTheDocument();
    
    });

});