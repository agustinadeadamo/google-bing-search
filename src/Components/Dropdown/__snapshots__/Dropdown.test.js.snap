// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Dropdown Component] Matches snapshot 1`] = `[Function]`;
